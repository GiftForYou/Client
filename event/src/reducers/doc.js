export const initialState = {
  doc_user:{},
  doc_detail:[],
  doc_recomendation:[]
};

export default (state = initialState, action) => {
  switch (action.type) {
    case 'SET_DOC_USER':
      return Object.assign({}, state, {doc_user: action.value});
    case 'SET_DOC_DETAIL':
      return Object.assign({}, state, {doc_detail: action.value});
    case 'SET_RECOMEND':
      return Object.assign({}, state, {doc_recomendation: action.value});
    default:
      return state;
  }
};
