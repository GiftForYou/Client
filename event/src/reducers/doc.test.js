import React from 'react';
import doc,{initialState} from './doc'

test('[Event Test] overAll initialState', () => {
  expect(initialState).toMatchSnapshot();
});

test('[Event Test] initialState have doc_user', () => {
  expect(initialState.doc_user).toMatchSnapshot();
});

test('[Event Test] initialState have doc_detail', () => {
  expect(initialState.doc_detail).toMatchSnapshot();
});

test('[Event Test] initialState have doc_recomendation', () => {
  expect(initialState.doc_recomendation).toMatchSnapshot();
});

test('[Event Test] doc empty', () => {
  expect(doc(initialState,{"action":{"type":''}})).toMatchSnapshot();
});

test('[Event Test] SET_DOC_USER', () => {
  expect(doc(initialState,{"action":{"type":'SET_DOC_USER',"value":'TEST'}})).toMatchSnapshot();
});

test('[Event Test] SET_DOC_DETAIL', () => {
  expect(doc(initialState,{"action":{"type":'SET_DOC_DETAIL',"value":'TEST'}})).toMatchSnapshot();
});
test('[Event Test] SET_RECOMEND', () => {
  expect(doc(initialState,{"action":{"type":'SET_RECOMEND',"value":'TEST'}})).toMatchSnapshot();
});
