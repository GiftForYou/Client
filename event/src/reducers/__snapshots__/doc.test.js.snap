// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Event Test] SET_DOC_DETAIL 1`] = `
Object {
  "doc_detail": Array [],
  "doc_recomendation": Object {
    "data": Object {
      "hasil": Array [],
    },
  },
  "doc_user": Object {},
  "status": false,
}
`;

exports[`[Event Test] SET_DOC_USER 1`] = `
Object {
  "doc_detail": Array [],
  "doc_recomendation": Object {
    "data": Object {
      "hasil": Array [],
    },
  },
  "doc_user": Object {},
  "status": false,
}
`;

exports[`[Event Test] SET_RECOMEND 1`] = `
Object {
  "doc_detail": Array [],
  "doc_recomendation": Object {
    "data": Object {
      "hasil": Array [],
    },
  },
  "doc_user": Object {},
  "status": false,
}
`;

exports[`[Event Test] doc empty 1`] = `
Object {
  "doc_detail": Array [],
  "doc_recomendation": Object {
    "data": Object {
      "hasil": Array [],
    },
  },
  "doc_user": Object {},
  "status": false,
}
`;

exports[`[Event Test] initialState have doc_detail 1`] = `Array []`;

exports[`[Event Test] initialState have doc_recomendation 1`] = `
Object {
  "data": Object {
    "hasil": Array [],
  },
}
`;

exports[`[Event Test] initialState have doc_user 1`] = `Object {}`;

exports[`[Event Test] overAll initialState 1`] = `
Object {
  "doc_detail": Array [],
  "doc_recomendation": Object {
    "data": Object {
      "hasil": Array [],
    },
  },
  "doc_user": Object {},
  "status": false,
}
`;
